extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -7,28 +7,131 @@
  * @package Jetpack
  */
 
-if (
-	( defined( 'IS_WPCOM' ) && IS_WPCOM ) ||
-	class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
-) {
-	jetpack_register_block(
-		'jetpack/tiled-gallery',
-		array(
-			'render_callback' => 'jetpack_tiled_gallery_load_block_assets',
-		)
-	);
+/**
+ * Jetpack Tiled Gallery Block class
+ *
+ * @since 7.3
+ */
+class Jetpack_Tiled_Gallery_Block {
+	/* Values for building srcsets */
+	const IMG_SRCSET_WIDTH_MAX  = 2000;
+	const IMG_SRCSET_WIDTH_MIN  = 600;
+	const IMG_SRCSET_WIDTH_STEP = 300;
 
 	/**
-	 * Tiled gallery block registration/dependency declaration.
+	 * Register the block
+	 */
+	public static function register() {
+		jetpack_register_block(
+			'jetpack/tiled-gallery',
+			array(
+				'render_callback' => array( __CLASS__, 'render' ),
+			)
+		);
+	}
+
+	/**
+	 * Tiled gallery block registration
 	 *
 	 * @param array  $attr    Array containing the block attributes.
 	 * @param string $content String containing the block content.
 	 *
 	 * @return string
 	 */
-	function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
+	public static function render( $attr, $content ) {
 		Jetpack_Gutenberg::load_assets_as_required( 'tiled-gallery' );
 
+		$is_squareish_layout = self::is_squareish_layout( $attr );
+
+		if ( preg_match_all( '/<img [^>]+>/', $content, $images ) ) {
+			/**
+			 * This block processes all of the images that are found and builds $find and $replace.
+			 *
+			 * The original img is added to the $find array and the replacement is made and added
+			 * to the $replace array. This is so that the same find and replace operations can be
+			 * made on the entire $content.
+			 */
+			$find    = array();
+			$replace = array();
+
+			foreach ( $images[0] as $image_html ) {
+				if (
+					preg_match( '/data-width="([0-9]+)"/', $image_html, $img_height )
+					&& preg_match( '/data-height="([0-9]+)"/', $image_html, $img_width )
+					&& preg_match( '/src="([^"]+)"/', $image_html, $img_src )
+				) {
+					// Drop img src query string so it can be used as a base to add photon params
+					// for the srcset.
+					$src_parts   = explode( '?', $img_src[1], 2 );
+					$orig_src    = $src_parts[0];
+					$orig_height = absint( $img_height[1] );
+					$orig_width  = absint( $img_width[1] );
+
+					// Because URLs are already "photon", the photon function used short-circuits
+					// before ssl is added. Detect ssl and add is if necessary.
+					$is_ssl = false !== strpos( $src_parts[1], 'ssl=1' );
```

Comment:
I'm getting `Notice: Undefined offset: 1 in /srv/users/user3d9fca1a/apps/user3d9fca1a/public/wp-content/plugins/jetpack-dev/extensions/blocks/tiled-gallery/tiled-gallery.php on line 72` for this.

Could we use `wp_parse_url` here and specifically target the `query` component, then check if `ssl=1` is present if the `src_query_args` is non null?

Reply:
That said, in the editor, it isn't saving with ssl=1, so probably need to fix this elsewhere from my quick read of the code.

Reply:
@kraftbj I pushed a simple fix for now; I do like the idea of using `parse_url` tho.